vms/platformvm/txs/mempool/mempool_test.go

```diff
@@ -0,0 +1,872 @@
+						BaseTx: avax.BaseTx{
+							Ins: []*avax.TransferableInput{
+								{
+									UTXOID: avax.UTXOID{
+										TxID: ids.ID{2},
+									},
+									Asset: avax.Asset{
+										ID: snowtest.AVAXAssetID,
+									},
+									In: &secp256k1fx.TransferInput{
+										Amt: 1,
+									},
+								},
+							},
+						},
+					},
+					TxID: ids.ID{2},
+				},
+			},
+			conflictsToRemove: set.Of(
+				ids.ID{1},
+			),
+			wantTxs: []ids.ID{{2}},
+		},
+		{
+			name: "remove multiple conflicts in mempool",
+			txs: []*txs.Tx{
+				{
+					Unsigned: &txs.BaseTx{
+						BaseTx: avax.BaseTx{
+							Ins: []*avax.TransferableInput{
+								{
+									UTXOID: avax.UTXOID{
+										TxID: ids.ID{1},
+									},
+									Asset: avax.Asset{
+										ID: snowtest.AVAXAssetID,
+									},
+									In: &secp256k1fx.TransferInput{
+										Amt: 1,
+									},
+								},
+							},
+						},
+					},
+					TxID: ids.ID{1},
+				},
+				{
+					Unsigned: &txs.BaseTx{
+						BaseTx: avax.BaseTx{
+							Ins: []*avax.TransferableInput{
+								{
+									UTXOID: avax.UTXOID{
+										TxID: ids.ID{2},
+									},
+									Asset: avax.Asset{
+										ID: snowtest.AVAXAssetID,
+									},
+									In: &secp256k1fx.TransferInput{
+										Amt: 1,
+									},
+								},
+							},
+						},
+					},
+					TxID: ids.ID{2},
+				},
+			},
+			conflictsToRemove: set.Of(
+				ids.ID{1},
+				ids.ID{2},
+			),
+		},
+		{
+			name: "remove conflicts not in mempool",
+			txs: []*txs.Tx{
+				{
+					Unsigned: &txs.BaseTx{
+						BaseTx: avax.BaseTx{
+							Ins: []*avax.TransferableInput{
+								{
+									UTXOID: avax.UTXOID{
+										TxID: ids.ID{1},
+									},
+									Asset: avax.Asset{
+										ID: snowtest.AVAXAssetID,
+									},
+									In: &secp256k1fx.TransferInput{
+										Amt: 1,
+									},
+								},
+							},
+						},
+					},
+					TxID: ids.ID{1},
+				},
+				{
+					Unsigned: &txs.BaseTx{
+						BaseTx: avax.BaseTx{
+							Ins: []*avax.TransferableInput{
+								{
+									UTXOID: avax.UTXOID{
+										TxID: ids.ID{3},
+									},
+									Asset: avax.Asset{
+										ID: snowtest.AVAXAssetID,
+									},
+									In: &secp256k1fx.TransferInput{
+										Amt: 1,
+									},
+								},
+							},
+						},
+					},
+					TxID: ids.ID{2},
+				},
+			},
+			conflictsToRemove: set.Of(ids.ID{123}),
+			wantTxs:           []ids.ID{{1}, {2}},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+
+			m, err := New(
+				"",
+				gas.Dimensions{1, 1, 1, 1},
+				1_000_000,
+				snowtest.AVAXAssetID,
+				prometheus.NewRegistry(),
+			)
+			require.NoError(err)
+
+			for _, tx := range tt.txs {
+				require.NoError(m.Add(tx))
+			}
+
+			conflictsToRemove := set.Set[ids.ID]{}
+			for conflict := range tt.conflictsToRemove {
+				conflictsToRemove.Add(conflict.Prefix(0))
+			}
+
+			m.RemoveConflicts(conflictsToRemove)
```

Comment:
I feel like this code is pretty confusing, perhaps it would be better to change
```golang
conflictsToRemove set.Set[ids.ID]
```
into
```golang
conflictsToRemove []avax.UTXOID
```

and then here it would look something like:
```suggestion
			conflictsToRemove := set.Set[ids.ID]{}
			for _, conflict := range tt.conflictsToRemove {
				conflictsToRemove.Add(conflict.InputID())
			}

			m.RemoveConflicts(conflictsToRemove)
```